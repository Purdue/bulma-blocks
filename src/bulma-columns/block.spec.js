jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickCheckbox,
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Columns', blockName: 'bulma-blocks/columns'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )
})