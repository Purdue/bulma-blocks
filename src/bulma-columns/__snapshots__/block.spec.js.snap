// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Bulma - Columns Block 🔎 Block should be available. 1`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Adding/Removing Columns 🔎 Add Columns. 1`] = `
"<!-- wp:bulma-blocks/columns {\\"numColumns\\":3,\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Adding/Removing Columns 🔎 Add then remove Columns. 1`] = `
"<!-- wp:bulma-blocks/columns {\\"numColumns\\":2,\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Adding/Removing Columns 🔎 Entering number larger than max should produce max number of columns 1`] = `
"<!-- wp:bulma-blocks/columns {\\"numColumns\\":6,\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column -->

<!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Checkbox Settings 🔎 Add Diviers Between Columns 1`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"dividers\\":true} -->
<div class=\\"columns is-multiline   has-dividers\\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Checkbox Settings 🔎 Center Columns 1`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":true} -->
<div class=\\"columns is-centered is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Background Color Options 1`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"bgColor\\":\\"has-background-black\\"} -->
<div class=\\"columns is-multiline  has-background-black \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Background Color Options 2`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"bgColor\\":\\"has-background-grey-lighter\\"} -->
<div class=\\"columns is-multiline  has-background-grey-lighter \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Background Color Options 3`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"bgColor\\":\\"has-background-black-ter\\"} -->
<div class=\\"columns is-multiline  has-background-black-ter \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Background Color Options 4`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Collapse Options 1`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"collapse\\":\\"is-mobile\\"} -->
<div class=\\"columns is-multiline is-mobile  \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Collapse Options 2`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false,\\"collapse\\":\\"is-desktop\\"} -->
<div class=\\"columns is-multiline is-desktop  \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;

exports[`🔬 Bulma - Columns Block 🔬 Side Panel Settings 🔬 Dropdown Settings 🔎 Columns Collapse Options 3`] = `
"<!-- wp:bulma-blocks/columns {\\"isCentered\\":false} -->
<div class=\\"columns is-multiline   \\"><!-- wp:bulma-blocks/column -->
<div class=\\"column     \\"></div>
<!-- /wp:bulma-blocks/column --></div>
<!-- /wp:bulma-blocks/columns -->"
`;
